src/main/java/edu/harvard/iq/dataverse/dataset/DatasetUtil.java

```diff
@@ -463,8 +473,18 @@ public static InputStream getLogoAsInputStream(Dataset dataset) {
             }
 
             try {
-                in = ImageThumbConverter.getImageThumbnailAsInputStream(thumbnailFile.getStorageIO(),
-                        ImageThumbConverter.DEFAULT_DATASETLOGO_SIZE).getInputStream();
+                boolean origImageFailed = thumbnailFile.isPreviewImageFail();
+                InputStreamIO isIO = ImageThumbConverter.getImageThumbnailAsInputStream(thumbnailFile.getStorageIO(),
+                        ImageThumbConverter.DEFAULT_DATASETLOGO_SIZE);
+                if (!origImageFailed && thumbnailFile.isPreviewImageFail()) {
+                    // We found an older 0 length thumbnail. Newer image uploads will not have this issue.
+                    // Once cleaned up, this thumbnail will no longer have this issue
+                    // ImageThumbConverter fixed the DataFile
+                    // Now we need to update dataset since this is a bad logo
+                    DatasetServiceBean datasetService = CDI.current().select(DatasetServiceBean.class).get();
+                    datasetService.removeDatasetThumbnail(dataset);
```

Comment:
@stevenwinship Just to follow up on the slack conversation: 
 `datasetService.removeDatasetThumbnail(dataset);` does NOT, and is not supposed to, remove the cached thumbnails associated with a specific DataFile. We may want to come up with a better, and potentially less confusing name for the method. Something like `datasetService.removeDatasetThumbnail(dataset);`. Because that's what it does - the purpose of the method is to make sure the Dataset no longer has a designated thumbnail. If that designated thumbnail was a "custom logo", it will attempt to delete the physical file (because, once we stop using it as the dataset thumbnail, there's no other use for it); but if it was one of the DataFile-level thumbnail, it will only clear the dataset-level  thumbnail designation, but it normally would not have any reason to attempt to delete the caches, because normally they would still be useful elsewhere on the page. 

I do believe that it is entirely safe to just leave the code above as is; I don't think you need to do anything special to delete these 0-size caches here, in the dataset-level thumbnail method; instead, those should be deleted in the datafile-level method, in ```ImageThumbConverter.getImageThumbnailAsInputStream(thumbnailFile.getStorageIO(),ImageThumbConverter.DEFAULT_DATASETLOGO_SIZE);```
where you are checking for that ```size == 0```. 

I hope this makes sense!

Reply:
Please note that I corrected an important typo in the above. When I was talking about changing the method name, I meant to say I wanted to name it ```clearDatasetThumbnail()```, instead of ```remove...``` 
I just typed the same name twice earlier ...

Reply:
(I'll go ahead and do that)